game/scripts/vscripts/items/drums_of_endurance.lua

```diff
@@ -0,0 +1,262 @@
+  return DOTA_UNIT_TARGET_HERO + DOTA_UNIT_TARGET_BASIC
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura:GetModifierAura()
+  return "modifier_item_drums_of_endurance_oaa_swiftness_aura_effect"
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura:IsAura()
+  return true
+end
+
+------------------------------------------------------------------------------------------------------------------------------
+--Aura_Modifier_Effect
+
+modifier_item_drums_of_endurance_oaa_swiftness_aura_effect = class(ModifierBaseClass)
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura_effect:OnCreated()
+  self.aura_movement_speed = self:GetAbility():GetSpecialValueFor("bonus_aura_movement_speed")
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura_effect:IsHidden()
+  return false
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura_effect:IsPurgable()
+  return false
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura_effect:IsDebuff()
+  return false
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura_effect:DeclareFunctions()
+  local decFuncs = {
+    MODIFIER_PROPERTY_MOVESPEED_BONUS_CONSTANT
+  }
+  return decFuncs
+end
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura_effect:GetModifierMoveSpeedBonus_Constant()
+  return self.aura_movement_speed
+end
+
+------------------------------------------------------------------------------------------------------------------------------
+-- Stats modifier
+
+modifier_item_drums_of_endurance_oaa = class(ModifierBaseClass)
+
+function modifier_item_drums_of_endurance_oaa:OnCreated()
+  self.bonus_intellect = self:GetAbility():GetSpecialValueFor("bonus_int")
+  self.bonus_strength = self:GetAbility():GetSpecialValueFor("bonus_str")
+  self.bonus_agility = self:GetAbility():GetSpecialValueFor("bonus_agi")
+  self.bonus_damage = self:GetAbility():GetSpecialValueFor("bonus_damage")
+  self.bonus_mana_regeneration = self:GetAbility():GetSpecialValueFor("bonus_mana_regen")
+
+  if IsServer() then
+    --If no previous drums aura then add the aura effect
+    local parent = self:GetParent()
+    -- Remove effect modifiers from units in radius to force refresh
+    local units = FindUnitsInRadius(
+      parent:GetTeamNumber(),
+      parent:GetAbsOrigin(),
+      nil,
+      self:GetAbility():GetSpecialValueFor("radius"),
+      DOTA_UNIT_TARGET_TEAM_FRIENDLY,
+      bit.bor(DOTA_UNIT_TARGET_BASIC, DOTA_UNIT_TARGET_HERO),
+      DOTA_UNIT_TARGET_FLAG_NONE,
+      FIND_ANY_ORDER,
+      false
+    )
+    if not self:GetCaster():HasModifier("modifier_item_drums_of_endurance_oaa_swiftness_aura") then
+      self:GetCaster():AddNewModifier(self:GetCaster(), self:GetAbility(), "modifier_item_drums_of_endurance_oaa_swiftness_aura", {})
+    end
+
+    local function RemoveEnduranceActive(unit)
+      unit:RemoveModifierByName("modifier_item_drums_of_endurance_oaa_active")
```

Comment:
You're deleting the wrong modifier.

Reply:
Also, I'm not sure, I might be guilty of doing this in some places too, but I think these would actually be clearer written in loop form rather than using the functional style.